src/CommonITILValidation.php

```diff
@@ -311,6 +311,21 @@ public function post_addItem()
         // Handle uploaded documents
         $this->input = $this->addFiles($this->input);
 
+        if (
+            isset($this->input['itemtype'])
+            && $this->input['itemtype'] == "Ticket"
+            && $this->input['pending'] == 1
+        ) {
+            $object     = new Ticket();
+            $object->getFromDB($this->input['tickets_id']);
+            $update['status'] = CommonITILObject::WAITING;
+            $update['id']     = $object->fields['id'];
+            // don't notify on ITILObject - update event
+            $update['_disablenotif'] = true;
+            // Use update method for history
+            $object->update($update);
+        }
+
```

Comment:
I think you should be able to reuse what we are already doing for followup and tasks here in this method:

![image](https://github.com/glpi-project/glpi/assets/42734840/b9bc8fab-aa53-4ff5-8df2-35475419aec1)

(Don't forget to add the `use Glpi\Features\ParentStatus;` statement).

And you will also need to handle pending reason edition in the `post_updateItem` function, like this:

![image](https://github.com/glpi-project/glpi/assets/42734840/f69c6a8d-0923-4740-a253-56d735e13ea0)